pyreisejl/utility/converters.py

```diff
@@ -1,132 +1,51 @@
-            "to_lat",
-            "to_lon",
-            "from_zone_id",
-            "to_zone_id",
-            "from_zone_name",
-            "to_zone_name",
-            "branch_device_type",
-        ],
-        inplace=True,
-    )
-    mpc["mpc"]["branch"] = branch.values
-    mpc["mpc"]["branchid"] = branchid
-    mpc["mpc"]["branchdevicetype"] = branchdevicetype
-
-    # generation cost
-    gencost = grid.gencost.copy()
-    gencost["before"].reset_index(inplace=True, drop=True)
-    gencost["before"].drop(columns=["interconnect"], inplace=True)
-    mpc["mpc"]["gencost"] = gencost["before"].values
-
-    # DC line
-    if len(grid.dcline) > 0:
-        dcline = grid.dcline.copy()
-        dclineid = dcline.index.values[np.newaxis].T
-        dcline.reset_index(inplace=True, drop=True)
-        dcline.drop(columns=["from_interconnect", "to_interconnect"], inplace=True)
-        mpc["mpc"]["dcline"] = dcline.values
-        mpc["mpc"]["dclineid"] = dclineid
-
-    # energy storage
-    mpc_storage = None
-
-    if len(grid.storage["gen"]) > 0:
-        storage = grid.storage.copy()
-
-        mpc_storage = {
-            "storage": {
-                "xgd_table": np.array([]),
-                "gen": np.array(storage["gen"].values, dtype=np.float64),
-                "sd_table": {
-                    "colnames": storage["StorageData"].columns.values[np.newaxis],
-                    "data": storage["StorageData"].values,
-                },
-            }
-        }
-
-    if filepath is not None:
-        savemat(filepath, mpc, appendmat=False)
-        if mpc_storage is not None:
-            savemat(storage_filepath, mpc_storage, appendmat=False)
-
-    return mpc, mpc_storage
+import os
+import pickle
+
+cols = {
+    "branch": ["branch_id", "from_bus_id", "to_bus_id", "x", "rateA"],
+    "dcline": ["dcline_id", "from_bus_id", "to_bus_id", "Pmin", "Pmax"],
+    "bus": ["bus_id", "Pd", "zone_id"],
+    "plant": [
+        "plant_id",
+        "bus_id",
+        "status",
+        "Pmin",
+        "Pmax",
+        "type",
+        "ramp_30",
+        "GenFuelCost",
+        "GenIOB",
+        "GenIOC",
+        "GenIOD",
+    ],
+}
+
+drop_cols = {"gencost": ["plant_id", "interconnect"]}
+drop_cols["gencost_orig"] = drop_cols["gencost"]
+
+
+def _save(path, name, df):
+    df = df.reset_index()
+    df = df.loc[:, cols.get(name, df.columns)]
+    df = df.drop(drop_cols.get(name, []), axis=1)
+    df.to_csv(os.path.join(path, f"{name}.csv"), index=False)
+
+
+def _save_storage(path, name, df):
+    df.to_csv(os.path.join(path, f"{name}.csv"), index=False)
+
+
+def pkl_to_csv(path):
+    with open(os.path.join(path, "grid.pkl"), "rb") as f:
+        grid = pickle.load(f)
+    _save(path, "branch", grid.branch)
+    _save(path, "dcline", grid.dcline)
+    _save(path, "bus", grid.bus)
+    _save(path, "plant", grid.plant)
+    _save(path, "gencost_orig", grid.gencost["before"])
+    _save(path, "gencost", grid.gencost["after"])
```

Comment:
This is o ut of the scope of this PR but I would keep `before` and `after` for `gencost` as we have in the `Grid` object. Also I think it is more explicit.

Reply:
Done